cmd/upgrade/upgrade.go

```diff
@@ -53,13 +52,22 @@ func NewCmdVolumeUpgrade(rootCmd *cobra.Command) *cobra.Command {
 		Short:   "Upgrade Volumes, storage engines, and interfaces in openebs application",
 		Aliases: []string{"update"},
 		Run: func(cmd *cobra.Command, args []string) {
-			if !util.IsValidCasType(upgradeOpts.CasType) {
-				fmt.Fprintf(os.Stderr, "cas-type %s not supported\n", upgradeOpts.CasType)
-			} else if upgradeOpts.CasType == util.JivaCasType {
-				upgrade.InstantiateJivaUpgrade(upgradeOpts)
-			} else {
-				fmt.Println("Only Jiva upgrades are available at this point")
-				fmt.Println("To upgrade other cas-types follow: https://github.com/openebs/upgrade#upgrading-openebs-reources")
+			switch upgradeOpts.CasType {
+			case util.JivaCasType:
+				{
+					upgrade.InstantiateJivaUpgrade(upgradeOpts)
+					break
+				}
+			case util.CstorCasType:
+				{
+					upgrade.InstantiateCspcUpgrade(upgradeOpts)
+					break
+				}
```

Comment:
Had one doubt, since CSPC and Cstor jobs will be handled separately, filtering on the basis of cas-type may not work here. Any other approach that you would recommend to follow.

Reply:
Okay, having different labels for both type of cstor jobs might help(i.e. a label key besides the `castype` one).

A Cstor volume upgrade can be thought of as a combination of a Pool upgrade and a Volume upgrade.

Reply:
Got it.
I was worried with the command to fire to upgrade cspc pools.
current command is:
```
./k upgrade --cas-type cstor
```